src/compiler/checker.ts

```diff
@@ -39432,11 +39431,33 @@ namespace ts {
             }
         }
 
-        function isConstantMemberAccess(node: Expression): boolean {
-            return node.kind === SyntaxKind.Identifier ||
-                node.kind === SyntaxKind.PropertyAccessExpression && isConstantMemberAccess((node as PropertyAccessExpression).expression) ||
-                node.kind === SyntaxKind.ElementAccessExpression && isConstantMemberAccess((node as ElementAccessExpression).expression) &&
-                    isStringLiteralLike((node as ElementAccessExpression).argumentExpression);
+        function isConstantMemberAccess(node: Expression): node is ElementAccessExpression | PropertyAccessExpression {
+            const type = getTypeOfExpression(node);
+            if(!type.symbol) {
+                return false;
+            }
+
+            if (node.kind === SyntaxKind.PropertyAccessExpression) {
+                const expr = node as PropertyAccessExpression;
+
+                if (expr.expression.kind === SyntaxKind.Identifier) {
+                    return true;
+                }
+
+                return isConstantMemberAccess(expr.expression);
+            }
+
+            if (node.kind === SyntaxKind.ElementAccessExpression) {
+                const expr = node as ElementAccessExpression;
+
+                if (expr.expression.kind === SyntaxKind.Identifier && isStringLiteralLike(expr.argumentExpression)) {
+                    return true;
+                }
+
+                return isConstantMemberAccess(expr.expression);
+            }
+
+            return false;

```

Comment:
on reading this, I *think* the only change is to ensure that returning true always means that `node: AccessExpression`. If so, I don't think it's worth the refactor. I'd prefer to keep the code the way it was and just check that `node` has a valid type.

Reply:
Do you mean that `ElementAccessExpression | PropertyAccessExpression` and `AccessExpression` are equivalent?

Reply:
No, the original code returns true directly when the node is `SyntaxKind.Identifier`, which needs to be fixed.

Reply:
Can you give an example of an input that's correct now and incorrect on the old code?

`isConstantMemberAccess` is initially called only with an `AccessExpression` so the `Identifier` check will never initially be true.

Reply:
You are right.
After I revert the `isConstantMemberAccess` logic change, all the test cases pass.
It seems that the issue is only caused by `errorType`.